pkg/reconciler/awssnssource/reconciler.go

```diff
@@ -33,8 +32,8 @@ type Reconciler struct {
 	base       common.GenericServiceReconciler
 	adapterCfg *adapterConfig
 
-	// API clients
-	secretsCli func(namespace string) coreclientv1.SecretInterface
+	// SNS client interface
+	snsCg ClientGetter
```

Comment:
Instead of explicitly passing a Kubernetes client to the Reconciler, we now rely on a `ClientGetter` abstraction which contract is to return a SNS client however it sees fits.

The real implementation still requires a Kubernetes client under the hood (see `controller.go`), but test implementations return mocks which do not need to round trip to Kubernetes.